### PR TITLE
GH-3617 : Value parameters in ReactiveKafkaProducerTemplate should be nullable

### PR DESCRIPTION
**Pull request by jukekxm**
* Fixes : #3617 
* add @Nullable to parameter of send method
* Fix to allow tombstone records to be transferred.

### CODE REVIEW

**Review by sobychacko — CHANGES_REQUESTED** [review]
Could you add your name to the author list?

File: spring-kafka/src/main/java/org/springframework/kafka/core/reactive/ReactiveKafkaProducerTemplate.java [hunk 1]
```diff
@@ -33,6 +33,7 @@
 import reactor.kafka.sender.SenderRecord;
 import reactor.kafka.sender.SenderResult;
 import reactor.kafka.sender.TransactionManager;
+import reactor.util.annotation.Nullable;
```
[the remark this answers is not in the document]
Comment (kimjuke) [reply spring-kafka/src/main/java/org/springframework/kafka/core/reactive/ReactiveKafkaProducerTemplate.java]: Thanks for your review!
I modified the package as you mentioned.

**Comment by kimjuke** [comment]
> Could you add your name to the author list?

i added my name, Thank you.

**Review by sobychacko — APPROVED** [review]

**Comment by sobychacko** [comment]
PR is failing due to checkstyle errors. Can you fix those?
```
Error: eckstyle] [ERROR] /home/runner/work/spring-kafka/spring-kafka/spring-kafka/src/main/java/org/springframework/kafka/core/reactive/ReactiveKafkaProducerTemplate.java:29:1: 'org.springframework.lang.Nullable' should be separated from previous imports. [ImportOrder]

Error: eckstyle] [ERROR] /home/runner/work/spring-kafka/spring-kafka/spring-kafka/src/main/java/org/springframework/kafka/core/reactive/ReactiveKafkaProducerTemplate.java:30:1: Wrong order for 'reactor.core.publisher.Flux' import. [ImportOrder]
```

**Comment by jukekxm** [comment]
> PR is failing due to checkstyle errors. Can you fix those?
> 
> ```
> Error: eckstyle] [ERROR] /home/runner/work/spring-kafka/spring-kafka/spring-kafka/src/main/java/org/springframework/kafka/core/reactive/ReactiveKafkaProducerTemplate.java:29:1: 'org.springframework.lang.Nullable' should be separated from previous imports. [ImportOrder]
> 
> Error: eckstyle] [ERROR] /home/runner/work/spring-kafka/spring-kafka/spring-kafka/src/main/java/org/springframework/kafka/core/reactive/ReactiveKafkaProducerTemplate.java:30:1: Wrong order for 'reactor.core.publisher.Flux' import. [ImportOrder]
> ```

I fixed them. Thank You :)

**Comment by sobychacko** [comment]
@jukekxm Thanks for the PR. It is now merged upstream.

**Pull request merged**